modules/utxo-lib/src/bitgo/signature.ts

```diff
@@ -0,0 +1,267 @@
+/**
+ * @prettier
+ */
+import * as opcodes from 'bitcoin-ops';
+import * as makeDebug from 'debug';
+
+import * as script from '../script';
+import * as crypto from '../crypto';
+import * as ECPair from '../ecpair';
+import * as ECSignature from '../ecsignature';
+import { Network } from '../networkTypes';
+
+const defaultDebug = makeDebug('bitgo:utxolib:signature');
+
+export interface Input {
+  hash: Buffer;
+  index: number;
+  sequence: number;
+  witness: Buffer;
+  script: Buffer;
+  signScript: Buffer;
+}
+
+export interface Transaction {
+  network: Network;
+
+  ins: Input[];
+
+  hashForSignatureByNetwork(index: number, pubScript: Buffer, amount: number, hashType: number, isSegwit: boolean);
+}
+
+const inputTypes = [
+  'multisig',
+  'nonstandard',
+  'nulldata',
+  'pubkey',
+  'pubkeyhash',
+  'scripthash',
+  'witnesspubkeyhash',
+  'witnessscripthash',
+  'witnesscommitment',
+] as const;
+
+type InputType = typeof inputTypes[number];
+
+export interface ParsedSignatureScript {
+  isSegwitInput: boolean;
+  inputClassification: InputType;
+  signatures?: Buffer[];
+  publicKeys?: Buffer[];
+  pubScript?: Buffer;
+}
+
+/**
+ * Parse a transaction's signature script to obtain public keys, signatures, the sig script,
+ * and other properties.
+ *
+ * Only supports script types used in BitGo transactions.
+ *
+ * @param input
+ * @returns ParsedSignatureScript
+ */
+export function parseSignatureScript(input: Input): ParsedSignatureScript {
```

Comment:
see https://github.com/BitGo/BitGoJS/blob/693b02dab445623848c301cce7a0c6dcfc0bf003/modules/core/src/v2/coins/abstractUtxoCoin.ts#L1288

Reply:
I haven't checked (and `verifySignature`) closely but it looks like straight copies or close to it from the methods you've referenced. Would the next step then be to remove these referenced methods from `core` and use the ones you're adding here instead?

Reply:
yes